src/hardware/adc.rs

```diff
@@ -148,7 +204,38 @@ impl Adc {
         // Setup first filter configuration register. 10Hz data rate. Sinc5Sinc1 Filter. No postfilter.
         adc.write(
             ad7172::AdcReg::FILTCON0,
-            ad7172::Filtcon::Order::SINC5SINC1 | ad7172::Filtcon::Odr::ODR_10,
+            ad7172::Filtcon::Order::SINC5SINC1 | ad7172::Filtcon::Odr::ODR_1_25,
         );
+
+        // Re-apply (also set after ADC reset) SYNC_EN flag in gpio register for standard synchronization
+        adc.write(ad7172::AdcReg::GPIOCON, ad7172::Gpiocon::SyncEn::ENABLED);
+    }
+
+    /// Handle adc interrupt.
+    pub fn handle_interrupt(&mut self) -> (InputChannel, u32) {
+        let (current_phy, ch) = Self::SCHEDULE[self.current_position];
+
+        let (data, status) = self.adcs.read_data();
```

Comment:
A few helpful comments about what is really happening will help you and others a lot.
Mention the important preconditions. Say why you can `read_data()` like this without any CS setting.
And say how it leaves the system (postcondition).
Maybe mention why the sequence is exactly like this (important part is the clearing of the flag).